cmd/cluster-init/main_test.go

```diff
@@ -0,0 +1,135 @@
+package main
+
+import (
+	"fmt"
+	"os"
+	"path/filepath"
+	"testing"
+
+	"github.com/google/go-cmp/cmp"
+
+	"github.com/openshift/ci-tools/pkg/testhelper"
+)
+
+func TestInitClusterBuildFarmDir(t *testing.T) {
+	workingDir, err := os.Getwd()
+	if err != nil {
+		t.Fatalf("couldn't obtain working directory")
+	}
+	testdata := filepath.Join(workingDir, "testdata")
+	testCases := []struct {
+		name string
+		options
+		expectedError error
+	}{
+		{
+			name: "basic",
+			options: options{
+				clusterName: "newcluster",
+				releaseRepo: testdata,
+			},
+		},
+		{
+			name: "symlink exists",
+			options: options{
+				clusterName: "existingCluster",
+				releaseRepo: testdata,
+			},
+			expectedError: fmt.Errorf("failed to symlink common to ../common"),
+		},
+	}
+	for _, tc := range testCases {
+		buildDir := filepath.Join(testdata, "clusters", "build-clusters", tc.clusterName)
+		t.Run(tc.name, func(t *testing.T) {
+			err := initClusterBuildFarmDir(tc.options)
+			if diff := cmp.Diff(tc.expectedError, err, testhelper.EquateErrorMessage); diff != "" {
+				t.Fatalf("error does not match expectedError, diff: %s", diff)
+			}
+
+			if tc.expectedError == nil {
+				if _, err := os.Stat(buildDir); os.IsNotExist(err) {
+					t.Fatalf("build farm directory: %s was not created", buildDir)
+				}
+				for _, item := range []string{"common", "common_except_app.ci"} {
+					expectedDest := filepath.Join("..", item)
+					dest, err := os.Readlink(filepath.Join(buildDir, item))
+					if err != nil || dest != expectedDest {
+						t.Fatalf("item: %s was not symlinked to: %s", item, expectedDest)
+					}
+				}
+			}
+		})
+
+		t.Cleanup(func() {
```

Comment:
nit: register your cleanup first, or you this won't run when the test fails early